nova/api/openstack/compute/sap_admin_api.py

```diff
@@ -69,6 +72,46 @@ def in_cluster_vmotion(self, req, body):
             common.raise_http_conflict_for_instance_invalid_state(
                 state_error, 'in_cluster_vmotion', server_id)
 
+    @_register_endpoint('GET')
+    def usage(self, req):
+        """Call
+        """
+        context = req.environ['nova.context']
+        context.can(sap_policies.POLICY_ROOT % 'usage:list')
+        return self._usage(context)
+
+    def show_usage(self, req, id):
+        """Call
+        """
+        context = req.environ['nova.context']
+        context.can(sap_policies.POLICY_ROOT % 'usage:show')
+
+        return self._usage(context, project_id=id)
+
+    def _usage(self, context, project_id=None):
+        results = nova_context.scatter_gather_skip_cell0(
+            context, objects.InstanceList.get_counts_by_project_and_az,
+            project_id=project_id)
+
+        aggregated = {}
+        for result in results.values():
+            if nova_context.is_cell_failure_sentinel(result):
+                continue
```

Comment:
Should we somehow communicate that the information is probably incomplete, so Limes can take the decision to use the information or not?
There's also a setting in Nova that's supposed to let an operator decide what to do on cell failure during listing instances (ignore silently or fail loudly). Do you know if that setting is honored in `scatter_gather_skip_cell0`?
I fear that returning incomplete information might lead to bad results in Limes.

Reply:
I'll raise a `503` error for it now.

Reply:
It should only appear if rabbitmq or the nova-conductor are offline. The latter being rather unlikely, as we do a rolling restart here.